### PR TITLE
css(fix): update :read-only intro

### LINKED ISSUE #29140: `read-only` pseudo class behavior ambiguities in the documentation.
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/:read-only

### What specific section or headline is this issue about?

:read-only

### What information was incorrect, unhelpful, or incomplete?

The first paragraph states:

> The :read-only [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS) [pseudo-class](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes) represents an element (such as input or textarea) that is not editable by the user. 

However, this is inaccurate. The `read-only` pseudo _targets editable fields that do not accept text_. This can be verified by the following code:

``` html
<input type="checkbox" />
<input type="radio" />
<input type="text" />
```

``` css
input {
  outline: none;
}

input:read-write {
  outline: thick double blue;
}

input:read-only {
  outline: thick double red;
}
```

Screenshot 2023-09-15 at 1 21 25 PM

_Sample output_

### What did you expect to see?

I think this should be mentioned in the documentation. That read only has unexpected (at least from my perspective) behavior. This could be an ambiguity-related issue similar to what can occur with `in-range` or other input-type specific pseudos, but I believe the documentation could benefit from this clarity.

### Do you have any supporting links, references, or citations?

Not beyond the replicable code.

### Do you have anything more you want to share?

No.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/_colon_read-only`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/:read-only
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/_colon_read-only/index.md
* Last commit: https://github.com/mdn/content/commit/d3cdafcdb4d22e5c55771501e7c80451a96aa032
* Document last modified: 2023-03-12T06:37:36.000Z

</details>

### PR DESCRIPTION
**Pull request by OnkarRuikar**
- fix https://github.com/mdn/content/issues/29140

Updating proes as per the [discussion](https://github.com/mdn/content/issues/29140#issuecomment-1722129597).

Also, converted GitHub sample to local live sample.

### CODE REVIEW

File: files/en-us/web/css/_colon_read-only/index.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ browser-compat: css.selectors.read-only
 
 {{CSSRef}}
 
-The **`:read-only`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) represents an element (such as `input` or `textarea`) that is not editable by the user.
+The **`:read-only`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) pseudo-class selects elements (such as text `input` or `textarea`) that are not editable by the user. Elements on which the HTML attribute [`readonly`](/en-US/docs/Web/HTML/Attributes/readonly) doesn't have an effect(such as radio, checkboxes, and all other non-form elements) are also selected by the `:read-only` pseudo-class.
```
Comment (Josh-Cena) [inline files/en-us/web/css/_colon_read-only/index.md:10]:
```suggestion
The **`:read-only`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) selects elements (such as text `input` or `textarea`) that are not editable by the user. Elements on which the HTML attribute [`readonly`](/en-US/docs/Web/HTML/Attributes/readonly) doesn't have an effect (such as `radio`, `checkbox`, and all other non-form elements) are also selected by the `:read-only` pseudo-class.
```

I think it would also be helpful to say this is `:not(:read-write)`, as per the linked spec discussion.

Comment (Josh-Cena) [reply files/en-us/web/css/_colon_read-only/index.md]: Looking at that PR, I don't think the final change mentioned `:read-write` at all?

I agree the current text is good and we should keep it. I just think additional information about being complementary is useful.

File: files/en-us/web/css/_colon_read-only/index.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ browser-compat: css.selectors.read-only
 
 {{CSSRef}}
 
-The **`:read-only`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) selects elements (such as text `input` or `textarea`) that are not editable by the user. Elements on which the HTML attribute [`readonly`](/en-US/docs/Web/HTML/Attributes/readonly) doesn't have an effect (such as `radio`, `checkbox`, and all other non-form elements) are also selected by the `:read-only` pseudo-class.
+The **`:read-only`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) selects elements (such as text `input` or `textarea`) that are not editable by the user. Elements on which the HTML attribute [`readonly`](/en-US/docs/Web/HTML/Attributes/readonly) doesn't have an effect (such as `radio`, `checkbox`, and all other non-form elements) are also selected by the `:read-only` pseudo-class. In short, the `:read-only` and `:not(:read-write)` selectors are equivalent.
```
Comment (Josh-Cena) [inline files/en-us/web/css/_colon_read-only/index.md:10]:
```suggestion
The **`:read-only`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) selects elements (such as text `input` or `textarea`) that are not editable by the user. Elements on which the HTML attribute [`readonly`](/en-US/docs/Web/HTML/Attributes/readonly) doesn't have an effect (such as `radio`, `checkbox`, and all other non-form elements) are also selected by the `:read-only` pseudo-class. In fact, `:read-only` matches anything that {{cssxref(":read-write")}} doesn't match, making it equivalent to `:not(:read-write)`.
```

File: files/en-us/web/css/_colon_read-only/index.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ browser-compat: css.selectors.read-only
 
 {{CSSRef}}
 
-The **`:read-only`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) represents an element (such as `input` or `textarea`) that is not editable by the user.
+The **`:read-only`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) selects elements (such as text `input` or `textarea`) that are not editable by the user. Elements on which the HTML attribute [`readonly`](/en-US/docs/Web/HTML/Attributes/readonly) doesn't have an effect (such as `radio`, `checkbox`, and all other non-form elements) are also selected by the `:read-only` pseudo-class. In fact, `:read-only` matches anything that {{cssxref(":read-write")}} doesn't match, making it equivalent to `:not(:read-write)`.
```
Comment (chrisdavidmills) [inline files/en-us/web/css/_colon_read-only/index.md:10]:
```suggestion
The **`:read-only`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) selects elements (such as certain {{htmlelement("input")}} types and {{htmlelement("textarea")}}) that are not editable by the user. Elements on which the HTML attribute [`readonly`](/en-US/docs/Web/HTML/Attributes/readonly) doesn't have an effect (such as [`<input type="radio">`](/en-US/docs/Web/HTML/Element/input/radio), [`<input type="checkbox">`](/en-US/docs/Web/HTML/Element/input/checkbox), and all other non-form elements) are also selected by the `:read-only` pseudo-class. In fact, `:read-only` matches anything that {{cssxref(":read-write")}} doesn't match, making it equivalent to `:not(:read-write)`.
```

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
@OnkarRuikar looking good! Just some suggestions for the opening paragraph.

**Review by chrisdavidmills — APPROVED** [review]
@OnkarRuikar awesome; merging!

**Pull request merged**